google_analytics/ga_homepage_report.py

```diff
@@ -1,365 +1,534 @@
+    percent_fmt = workbook.add_format({'num_format': '0.0%', 'bold': False})
+    comma_fmt = workbook.add_format({'num_format': '#,##0', 'bold': False})
+    date_fmt = workbook.add_format({'num_format': 'dd/mm/yy'})
+    cell_format = workbook.add_format({'bold': True, 'italic': False})
+    merge_format = workbook.add_format(
+        {
+            'bold': 1,
+            'align': 'center',
+            'valign': 'vcenter',
+        }
+    )
+
+
+    # Create the homepage courses worksheet
+    if featured_cards is not None:
+        featured_cards.to_excel(writer, index=False, sheet_name='Featured Card Report', startrow=18)
+        featured_cards_worksheet = writer.sheets['Featured Card Report']
+
+        # Set conditional format
+        # featured_cards_worksheet.conditional_format('F1:F1000', {'type': '3_color_scale'})
+        # featured_cards_worksheet.conditional_format('D1:D1000', {'type': '3_color_scale'})
+        # featured_cards_worksheet.conditional_format('C1:C1000', {'type': '3_color_scale'})
+        # featured_cards_worksheet.conditional_format('G1:G1000', {'type': '3_color_scale'})
+
+        # Set column width and formatting
+        featured_cards_worksheet.set_column('A:A', 60)
+        featured_cards_worksheet.set_column('D:D', 15, comma_fmt)
+        featured_cards_worksheet.set_column('E:E', 15, percent_fmt)
+        featured_cards_worksheet.set_column('F:H', 15, comma_fmt)
+        featured_cards_worksheet.set_column('I:O', 15, percent_fmt)
+
+        # Write headings
+        # featured_cards_worksheet.write('A1', 'Homepage Course Enrollments, Data from '+str(start_date)+' to '+str(end_date) , cell_format)
+        # featured_cards_worksheet.write('A3', 'Total Homepage Views:', cell_format)
+        # featured_cards_worksheet.write('A4', 'Total Click-Throughs to Courses Below:', cell_format)
+        # featured_cards_worksheet.write('A5', 'Total CTR:', cell_format)
+        # featured_cards_worksheet.write('A6', 'Total Enrollments:', cell_format)
+        # featured_cards_worksheet.write('A7', 'Total Enrollments per Homepage View:', cell_format)
+
+        # # Write data
+        # featured_cards_worksheet.write('B3', total_homepage_views, comma_fmt)
+        # featured_cards_worksheet.write('B4', int(featured_cards['uniqueViews'].sum()), comma_fmt)
+        # featured_cards_worksheet.write('B5', float(featured_cards['uniqueViews'].sum())/total_homepage_views, percent_fmt)
+        # featured_cards_worksheet.write('B6', int(featured_cards['uniqueEnrolls'].sum()), comma_fmt)
+        # featured_cards_worksheet.write('B7', float(featured_cards['uniqueEnrolls'].sum())/total_homepage_views, percent_fmt)
+
+        featured_cards_worksheet.write(
+            'A1',
+            'Homepage Course Enrollments, Data from {start} to {end}'.format(start=start_date, end=end_date),
+            cell_format
+        )
+        featured_cards_worksheet.write('A3', 'Overview', cell_format)
```

Comment:
Well based on your comment that this is being sunsetted I don't know if we care about the code quality.
However, the value would be that you could define the things you're writing compactly in a couple of lines and only calling the statement fewer times would mean this block would be like 5 lines instead of 40